### PR TITLE
Fixes #34599 - CV UI - Wizard bug fixes

### PR DESCRIPTION
**Pull request by Andrewgdewar**
#### What are the changes introduced in this pull request?

- Delete contentView wizard should now make api calls on open if needed to determine version data, it should also redirect the user to the CV index page on completion if not already on the cv Index page.
- Fixed an issue on CvDetails where calling the cvdetails update call would cause double calls (unified cvDetailsContainer)
- Publish content View wizard, now directs the user to the CV Versions detail page if successful.

Fixed a number of issues created by using wizards in different locations (cvDelete on index vs CVHeader, Publish in versions details vs versions index page)

1. Fixed CV Index page, delete contentView wizard flashes and disappears due to page reload.
2. Fixed After deleting/publishing/promoting content views, some relevant APIs are not being updated.
3. Fixed ContentView Delete modal shows blank loading when on the following route '/content_views/<id>#/<anything other than versions>' if user has not traversed the versions page previously.
4. Fixed an issue where pollingTasks would be called infinitely on the cvIndex page.

#### What are the testing steps for this pull request?
- Perform crud actions from Wizards in different locations, keep an eye on the network tab, ensure that the user is directed to the correct locations and that all functionality is as expected.

#### Caveats
- Please help identify any other issues around navigation/data loading in other wizards if you see/think of any.

### CODE REVIEW

File: webpack/scenes/ContentViews/Publish/CVPublishFinish.js [hunk 1]
```diff
@@ -182,10 +176,9 @@ CVPublishFinish.propTypes = {
   setDescription: PropTypes.func.isRequired,
   userCheckedItems: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
   setUserCheckedItems: PropTypes.func.isRequired,
-  setIsOpen: PropTypes.func.isRequired,
+  onClose: PropTypes.func.isRequired,
```
[the remark this answers is not in the document]
Comment (Andrewgdewar) [reply webpack/scenes/ContentViews/Publish/CVPublishFinish.js:179]: Done

**Comment by Andrewgdewar** [comment]
> nent version.

Will address.

File: webpack/scenes/ContentViews/Delete/Steps/CVDeletionFinish.js [hunk 2]
```diff
@@ -21,53 +23,59 @@ const CVDeletionFinish = () => {
...
+      dispatch(removeContentViewVersion(
+        cvId, cvId, cvEnvironments, params,
+        // Only when we are on the contentViews page do we need to call getContentViews.
+        () => {
+          if (pathname === '/content_views') dispatch(getContentViews());
+          else push('/content_views');
+          setIsOpen(false);
+        },
+        () => {
+          setIsOpen(false);
```
Comment (sjha4) [inline webpack/scenes/ContentViews/Delete/Steps/CVDeletionFinish.js:71]: We do need to add this on publish/promote as well. Tested errors on those before planning and the wizard doesn't autoclose.

File: webpack/scenes/ContentViews/Details/Versions/Delete/RemoveSteps/CVVersionDeleteFinish.js [hunk 3]
```diff
@@ -15,50 +15,27 @@ const CVVersionDeleteFinish = () => {
...
-    if (removeResolved && removeCVVersionResponse && removeDispatched) {
-      dispatch(getContentViewVersions(cvId));
-      if (detailsPage) {
-        setRedirect(true);
-      } else {
-        setIsOpen(false);
-      }
+  useEffect(() => {
+    if (STATUS.ERROR) {
+      setIsOpen(false);
```
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/Versions/Delete/RemoveSteps/CVVersionDeleteFinish.js:30]: Is this doing a if (removeCVVersionStatus === STATUS.ERROR) check? We can have that specified. This feels like magic.. :D

Comment (Andrewgdewar) [reply webpack/scenes/ContentViews/Details/Versions/Delete/RemoveSteps/CVVersionDeleteFinish.js:30]: lol.. I think that was a typo.

File: webpack/scenes/ContentViews/Delete/Steps/CVDeletionFinish.js (hunk 2 above)
Comment (Andrewgdewar) [reply webpack/scenes/ContentViews/Delete/Steps/CVDeletionFinish.js:71]: will do, tests don't like this syntax by the way, so you may see alternatives (like in the removeFinish).

**Review by sjha4 — APPROVED** [review]
Ack 👍🏼

**Pull request merged**